contracts/StableJoeStaking.sol

```diff
@@ -0,0 +1,181 @@
+        uint256 rewardDebt; /// @notice The sum of rewards already paid to this account. See explanation below.
+        /**
+         * @notice We do some fancy math here. Basically, any point in time, the amount of JOEs
+         * entitled to a user but is pending to be distributed is:
+         *
+         *   pending reward = (user.amount * accTokenPerShare) - user.rewardDebt
+         *
+         * Whenever a user deposits or withdraws moJoe. Here's what happens:
+         *   1. accTokenPerShare (and `lastTokenBalance`) gets updated.
+         *   2. User receives the pending reward sent to his/her address.
+         *   3. User's `amount` gets updated.
+         *   4. User's `rewardDebt` gets updated.
+         */
+    }
+
+    /// @notice Info.
+    IERC20Upgradeable moJoe; /// @notice Address of moJoe contract.
+    uint256 lastTokenBalance; /// @notice Last balance of reward token.
+    uint256 accTokenPerShare; /// @notice Accumulated Tokens per share, times 1e12. See above.
+
+    /// @notice The reward token.
+    IERC20Upgradeable public rewardToken;
+
+    /// @notice Info of each user that stakes moJoe.
+    mapping(address => UserInfo) public userInfo;
+
+    /// @notice Emitted when a user deposits its moJOE
+    event Deposit(address indexed user, uint256 amount);
+
+    /// @notice Emitted when a user withdraws its moJOE
+    event Withdraw(address indexed user, uint256 amount);
+
+    /// @notice Emitted when a user emergency withdraws its moJOE
+    event EmergencyWithdraw(address indexed user, uint256 amount);
+
+    /**
+     * @notice Initialize a new StableJoeStaking contract.
+     * @dev This contract needs to receive an ERC20 `_rewardToken` in order to distribute them
+     * (with JoeMakerV4 in our case)
+     * @param _rewardToken The address of the ERC-20 reward token
+     * @param _moJoe The address of the ERC-20 moJOE token, traderjoe's staking token.
+     */
+    function initialize(IERC20Upgradeable _rewardToken, IERC20Upgradeable _moJoe) public initializer {
+        __Ownable_init();
+
+        rewardToken = _rewardToken;
+        moJoe = _moJoe;
+    }
+
+    /**
+     * @notice View function to see pending reward token on frontend.
+     * @param _user The address of the user
+     * @return `_user`'s pending reward token
+     */
+    function pendingToken(address _user) external view returns (uint256) {
+        UserInfo storage user = userInfo[_user];
+        uint256 lpSupply = moJoe.balanceOf(address(this));
+        uint256 _accTokenPerShare = accTokenPerShare;
+
+        uint256 rewardTokenBalance = rewardToken.balanceOf(address(this));
+        if (rewardTokenBalance != lastTokenBalance && lpSupply != 0) {
+            uint256 rewardTokenAdded = rewardTokenBalance.sub(lastTokenBalance);
+            _accTokenPerShare = _accTokenPerShare.add(rewardTokenAdded.mul(1e12).div(lpSupply));
+        }
+        return user.amount.mul(_accTokenPerShare).div(1e12).sub(user.rewardDebt);
```

Comment:
nit: suggest to put `1e12` into a `public constant` and used that instead since its used multiple times in this contract